fs/fs_zenfs.cc

```diff
@@ -277,8 +277,8 @@ IOStatus ZenFS::WriteEndRecord(ZenMetaLog* meta_log) {
 
 /* Assumes the files_mtx_ is held */
 IOStatus ZenFS::RollMetaZoneLocked() {
-  ZenMetaLog* new_meta_log;
-  Zone *new_meta_zone, *old_meta_zone;
+  std::unique_ptr<ZenMetaLog> new_meta_log, old_meta_log;
```

Comment:
Do you think it would be better to split the metalog related changes to another patch ?

Reply:
I can do that, I have no opinion on that. But the system would be broken after application of the first patch, before the second one.